flutter/lib/src/navigation/sentry_navigator_observer.dart

```diff
@@ -262,22 +255,27 @@ class SentryNavigatorObserver extends RouteObserver<PageRoute<dynamic>> {
   }
 
   Future<void> _startTransaction(
-      Route<dynamic>? route, DateTime startTimestamp) async {
-    String? name = _getRouteName(route);
+    Route<dynamic>? route,
+    DateTime startTimestamp,
+  ) async {
+    final routeName = _getRouteName(route) ?? _currentRouteName;
     final arguments = route?.settings.arguments;
 
-    if (name == null || (name == '/')) {
+    final isRoot = routeName ==
+        '/'; // Root transaction is already created by the app start integration.
+    if (!_enableAutoTransactions || routeName == null || isRoot) {
       return;
     }
 
     final transactionContext = SentryTransactionContext(
-      name,
+      routeName,
       SentrySpanOperations.uiLoad,
       transactionNameSource: SentryTransactionNameSource.component,
       origin: SentryTraceOrigins.autoNavigationRouteObserver,
     );
+    _timeToDisplayTracker?.transactionId = transactionContext.spanId;
```

Comment:
I'm not sure if this entirely fixes the problem, since the function is still running as async there's a chacne this will still be executed at some point later rather than before the widget is built

Reply:
True, but the transaction context is sync, and the router didPush should always be called before. So we'd have the reference in any case. Only problem i'd see is if a user tries to immediatley call `reportFullDisplay` sync, as the transaction has not been started at this point. But then again, there's no point in using the API this way.

Reply:
but the `_startTransaction` function is not sync, I assume there could be edge cases where the event loop won't pick up this function in time and then a timing issue might occur again

I'm thinking it's safer to move the transaction context creation out of this async function and pass the context in _startTransaction as a parameter